src/data/information_panel.rs

```diff
@@ -0,0 +1,390 @@
+    pub fn add_metadata_loader(
+        mut self,
+        extension: &str,
+        load_metadata: impl FnMut(&mut IndexMap<String, String>, &PathBuf) + 'static,
+    ) -> Self {
+        self.additional_meta_files
+            .insert(extension.to_string(), Box::new(load_metadata));
+        self
+    }
+
+    /// Reads a preview of the file if it is detected as a text file.
+    fn load_text_file_preview(path: PathBuf, max_chars: usize) -> io::Result<String> {
+        let mut file = File::open(path)?;
+        let mut chunk = [0; 96]; // Temporary buffer
+        let mut buffer = String::new();
+
+        // Add the first chunk to the buffer as text
+        let mut total_read = 0;
+
+        // Continue reading if needed
+        while total_read < max_chars {
+            let bytes_read = file.read(&mut chunk)?;
+            if bytes_read == 0 {
+                break; // End of file
+            }
+            let chars_read: String = String::from_utf8(chunk[..bytes_read].to_vec())
+                .map_err(|_| io::Error::from(io::ErrorKind::InvalidData))?;
+            total_read += chars_read.len();
+            buffer.push_str(&chars_read);
+        }
+
+        Ok(buffer.to_string())
+    }
+    fn load_content(&self, path: PathBuf) -> Option<String> {
+        if self.load_text_content {
+            Self::load_text_file_preview(path, self.text_content_max_chars).ok()
+        } else {
+            None
+        }
+    }
+
+    /// Renders the Information Panel in the provided UI context.
+    ///
+    /// # Arguments
+    /// - `ui`: The UI context where the panel will be rendered.
+    /// - `file_dialog`: A reference to the current file dialog, which provides the active file entry.
+    pub fn ui(&mut self, ui: &mut Ui, file_dialog: &mut FileDialog) {
+        const SPACING_MULTIPLIER: f32 = 4.0;
+
+        ui.label("Information");
+        ui.separator();
+
+        // Display metadata in a grid format
+        let width = file_dialog.config_mut().right_panel_width.unwrap_or(100.0) / 2.0;
+
+        if let Some(item) = file_dialog.active_entry() {
+            // load file content and additional metadata if it's a new file
+            let path_option = file_dialog.active_entry();
+            if let Some(path) = path_option {
+                if self.loaded_file_name != path.to_path_buf() {
+                    self.loaded_file_name = path.to_path_buf();
+                    // clear previous meta data
+                    self.other_meta_data = IndexMap::default();
+                    if let Some(ext) = path.to_path_buf().extension() {
+                        if let Some(ext_str) = ext.to_str() {
+                            if let Some(load_meta_data) =
+                                self.additional_meta_files.get_mut(ext_str)
+                            {
+                                // load metadata
+                                load_meta_data(&mut self.other_meta_data, &path.to_path_buf());
```

Comment:
`path.to_path_buf` is used very often here, which creates many unnecessary copies of the path. It is enough if we call `.to_path_buf` once and save it in a variable.